src/couchdb/setup-database.ts

```diff
@@ -90,4 +108,66 @@ export async function setupDatabase(
       }
     })
   }
+
+  // Set up replication:
+  if (replicatorSetup != null && currentCluster != null) {
+    // Figure out the current username:
+    const sessionInfo = await connection.session()
+    const currentUsername: string = sessionInfo.userCtx.name
+
+    // Helper to create replication documents on demand:
+    const setupReplicator = async (): Promise<void> => {
+      const { replicators } = replicatorSetup.doc
+      if (replicators[currentCluster] == null) return
+
+      function makeEndpoint(clusterName: string): ReplicatorEndpoint {
+        const row = replicators[clusterName]
+        const url = `${row.url.replace(/[/]$/, '')}/${name}`
+        return row.basicAuth == null
+          ? url
+          : { url, headers: { Authorization: `Basic ${row.basicAuth}` } }
+      }
+
+      const documents: { [name: string]: ReplicatorDocument } = {}
+      for (const remoteCluster of Object.keys(replicators)) {
+        if (remoteCluster === currentCluster) continue
+        const { mode } = replicators[remoteCluster]
+
+        if (mode === 'source' || mode === 'both') {
+          documents[`${name}.from-${remoteCluster}`] = {
```

Comment:
Use `.` as the delimeter for this level.

Reply:
Done.